src/SFA.DAS.LevyTransferMatching.Web/Orchestrators/OpportunitiesOrchestrator.cs

```diff
@@ -96,12 +96,42 @@ public async Task<OpportunitySummaryViewModel> GetOpportunitySummaryViewModel(Op
             return new OpportunitySummaryViewModel()
             {
                 Amount = opportunityDto.Amount,
-                Description = opportunityDto.IsNamePublic ? $"{opportunityDto.DasAccountName} ({encodedPledgeId})" : "A levy-paying business wants to fund apprenticeship training in:",
+                Description = GenerateDescription(opportunityDto, encodedPledgeId),
                 JobRoleList = jobRoleList,
                 LevelList = levelList,
                 SectorList = sectorList,
                 YearDescription = dateTime.ToTaxYearDescription(),
             };
         }
+
+        public async Task<ApplyViewModel> GetApplyViewModel(string encodedPledgeId)
+        {
+            var opportunityDto = await _opportunitiesService.GetOpportunity((int)_encodingService.Decode(encodedPledgeId, EncodingType.PledgeId));
```

Comment:
You can decode this using the [AutoDecode] attribute on the request itself - example:

```
        [AutoDecode(nameof(EncodedAccountId), EncodingType.AccountId)]
        public long AccountId { get; set; }
```

Reply:
That's awesome!